setup/install/providers/azure.md

```diff
@@ -79,11 +85,19 @@ hal config provider azure account add my-azure-account \
   --subscription-id $SUBSCRIPTION_ID \
   --default-key-vault $VAULT_NAME \
   --default-resource-group $RESOURCE_GROUP \
+  --packer-resource-group $RESOURCE_GROUP \
   --app-key
 ```
 
-> NOTE: You will be prompted for the App Key on standard input. If necessary,
-you can generate a new key: `az ad sp credential reset --name $APP_ID`
+> NOTE:
+> 1. You will be prompted for the App Key on standard input. If necessary,
+you can generate a new key: `az ad sp credential reset --name $APP_ID`<br/>
+> 2. Starting from Halyard v1.19, SSH public key will be used to provision VM scale set by default. If you prefer to use password, add the parameter<br/>
+`--useSshPublicKey "false" \`<br/>
+Prior to Halyard v1.19, the default credential option is password and halyard does not support the above parameter to switch the option to SSH public key.<br/>
```

Comment:
If before 1.19 Halyard doesn't *support* the SSH public key option, is it correct to say that password is the *default* credential option?

Reply:
Good catch! The justification is:
1. Prior to 1.19, the default behavior is password, but there is an option to enable ssh public key at the cloud driver side with some manual config, though hal does not support set that.
2. As for users that are stick to prior-hal-v.1.19 and need the SSH key solution, we have an offline document for instruction. Indeed, we have offline sent that doc to some customers that belong to this cohort;
3. I personally do not prefer to put the content of the doc mentioned at #2 here given this is an official doc and that instruction bypassing hal is a workaround. On the other hand, I assume things could become better once hal v1.19 comes into being in the near future;
4. I do not prefer to make a solid statement here say hal prior-v1.19 does not support SSH key, because we do support that and that statement may keep some of our potential users away. 
With these, feel free to suggest any proposals and I'm willing to follow.